### PR TITLE
Category page configuration

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds category page configuration state, selector, backend call, and transformer such as the current page number, number of products per page, and filtering and sorting options.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lderrickable** [comment]
I want to do this first PR to make sure the base stuff is good, then I'll do a small PR for when changing filters, sort_mode, page_size, or sort_direction resets the current_page to 1.

File: libs/category/src/reducers/category/category.reducer.ts [hunk 1]
```diff
@@ -2,17 +2,51 @@ import { DaffCategoryActionTypes, DaffCategoryActions } from '../../actions/cate
 import { CategoryReducerState } from './category-reducer-state.interface';
 
 const initialState: CategoryReducerState = {
-  selectedCategoryId: null,
+  categoryPageConfigurationState: {
+    id: null,
```
Comment (damienwebdev) [inline libs/category/src/reducers/category/category.reducer.ts]: Remove the `id`

File: libs/category/src/reducers/category/category.reducer.ts [hunk 2]
```diff
@@ -2,17 +2,51 @@ import { DaffCategoryActionTypes, DaffCategoryActions } from '../../actions/cate
...
   loading: false,
   errors: []
 };
 
 export function categoryReducer(state = initialState, action: DaffCategoryActions): CategoryReducerState {
   switch (action.type) {
     case DaffCategoryActionTypes.CategoryLoadAction:
-      return { ...state, loading: true };
+      return { 
+        ...state,
```
Comment (damienwebdev) [inline libs/category/src/reducers/category/category.reducer.ts]: Document why these two state changes are different.

File: libs/category/src/models/category-page-configuration-state.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { DaffCategoryFilter } from './category-filter';
+import { DaffCategorySortOption } from './category-sort-option';
+import { DaffCategoryRequest } from './category-request';
+
+export interface DaffCategoryPageConfigurationState extends DaffCategoryRequest {
```
Comment (damienwebdev) [inline libs/category/src/models/category-page-configuration-state.ts]: This should extend `Omit<DaffCategoryRequest, 'id'>`

**Comment by damienwebdev** [comment]
This generally LGTM.

File: libs/category/src/reducers/category/category.reducer.ts [hunk 4]
```diff
@@ -2,17 +2,51 @@ import { DaffCategoryActionTypes, DaffCategoryActions } from '../../actions/cate
...
 export function categoryReducer(state = initialState, action: DaffCategoryActions): CategoryReducerState {
   switch (action.type) {
     case DaffCategoryActionTypes.CategoryLoadAction:
-      return { ...state, loading: true };
+      return { 
+        ...state, 
+        loading: true,
+        categoryPageConfigurationState: {
+          ...state.categoryPageConfigurationState,
+          id: action.payload.id,
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply libs/category/src/reducers/category/category.reducer.ts]: After thinking about this a bit more, we need the `id`.

File: libs/category/src/drivers/magento/category.service.ts [hunk 5]
```diff
@@ -23,16 +25,24 @@ export class DaffMagentoCategoryService implements DaffCategoryServiceInterface
...
+      this.apollo.query<GetACategoryResponse>(this.queryManager.getACategoryQuery(
+        parseInt(categoryRequest.id, 10)
+      )),
+      this.apollo.query<any>(this.productQueryManager.getSortFieldsAndFiltersByCategory(categoryRequest.id))
+    ]).pipe(
       map(result => {
         return {
-          category: this.magentoCategoryTransformerService.transform(result.data.category),
-          products: this.magentoProductTransformerService.transformMany(result.data.category.products.items)
+          category: this.magentoCategoryTransformerService.transform(result[0].data.category),
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/category/src/drivers/magento/category.service.ts]: done

File: libs/category/src/reducers/category/category.reducer.ts (hunk 2 above)
Comment (lderrickable) [reply libs/category/src/reducers/category/category.reducer.ts]: done

**Comment by lderrickable** [comment]
I also made some changes to the models for product filters. I had some in the category module, but realized they need to go in product

**Pull request merged**